src/frontend/js/hooks/useUserWishlistCourses/index.spec.tsx

```diff
@@ -0,0 +1,188 @@
+      wrapper: Wrapper,
+    });
+
+    await waitFor(() => expect(result.current.states.fetching).toBe(true));
+    responseDeferred.resolve(wishlistCourse);
+
+    await waitFor(() => expect(result.current.states.fetching).toBe(false));
+    expect(fetchMock.called(urlGetWishlistById)).toBeTruthy();
+    expect(result.current.states.error).toBe(undefined);
+    expect(JSON.stringify(result.current.item)).toBe(JSON.stringify(wishlistCourse));
+  });
+
+  it('retrieves wishlist cours from courseCode', async () => {
+    const responseDeferred = new Deferred();
+    const urlGetWishlistByFilters = `https://joanie.test/api/v1.0/wishlist/?courseCode=${wishlistCourse.course_code}`;
+    fetchMock.get(urlGetWishlistByFilters, responseDeferred.promise);
+
+    const { result } = renderHook(
+      () =>
+        useUserWishlistCourses({
+          courseCode: wishlistCourse.course_code,
+        }),
+      {
+        wrapper: Wrapper,
+      },
+    );
+    await waitFor(() => expect(result.current.states.fetching).toBe(true));
+
+    responseDeferred.resolve(wishlistCourse);
+    await waitFor(() => expect(result.current.states.fetching).toBe(false));
+
+    expect(fetchMock.called(urlGetWishlistByFilters)).toBeTruthy();
+    expect(result.current.states.error).toBe(undefined);
+    expect(JSON.stringify(result.current.items)).toBe(JSON.stringify([wishlistCourse]));
+  });
+
+  it('retrieves full user wishlist of cours', async () => {
+    const wishlistList = mockFactories.UserWishlistCourseFactory.generate(3);
+    const responseDeferred = new Deferred();
+
+    const urlGetWishlistFull = 'https://joanie.test/api/v1.0/wishlist/';
+    fetchMock.get(urlGetWishlistFull, responseDeferred.promise);
+
+    const { result } = renderHook(() => useUserWishlistCourses(), {
+      wrapper: Wrapper,
+    });
+
+    await waitFor(() => expect(result.current.states.fetching).toBe(true));
+    responseDeferred.resolve(wishlistList);
+
+    await waitFor(() => expect(result.current.states.fetching).toBe(false));
+    expect(fetchMock.called(urlGetWishlistFull)).toBeTruthy();
+    expect(result.current.states.error).toBe(undefined);
+    expect(JSON.stringify(result.current.items)).toBe(JSON.stringify(wishlistList));
+  });
+
+  it.only('add a cours to user wishlist', async () => {
+    const getAllResponseDeferred = new Deferred();
+    const urlWishlist = `https://joanie.test/api/v1.0/wishlist/`;
+    fetchMock.get(urlWishlist, getAllResponseDeferred.promise);
+    const { result } = renderHook(() => useUserWishlistCourses(), {
+      wrapper: Wrapper,
+    });
+    getAllResponseDeferred.resolve([]);
+    await waitFor(() => expect(result.current.states.fetching).toBe(false));
+    fetchMock.post(urlWishlist, wishlistCourse);
+
+    const createdUserWishlistCourse = await result.current.methods.create(
+      wishlistCourse.course_code,
+    );
+    await waitFor(() => expect(createdUserWishlistCourse).not.toBeUndefined());
```

Comment:
Je ne comprend pas pourquoi je n'ai pas le retour de mon mock ici: 
```shell
expect(received).not.toBeUndefined()
Received: undefined
```

Reply:
That's mainly because `create` method uses `mutate` method exposed by `useMutation` under the hood and `mutate` method is not async.

With this method, the only way to retrieve data return is through the `onSuccess` callback.

In your case, a workaround could be :
```ts
let createdUserWishlistCourse: UserWishlistCourse;
    result.current.methods.create(wishlistCourse.course_code, {
      onSuccess: (data) => (createdUserWishlistCourse = data),
    });
    await waitFor(() => expect(createdUserWishlistCourse).not.toBeUndefined());
```

Reply:
`useResourcesRoot` only handle remote resources, ,xould we use `asyncMutate` instead of `mutate` ?
By using `mutate` here the states `creating`, `deleting`, etc are always false and therefor useless.